flake.nix

```diff
@@ -1,34 +1,54 @@
 {
   inputs.flake-utils.url = github:numtide/flake-utils;
 
-  nixConfig = {
-    # Needed by callCabal2nix
-    allow-import-from-derivation = true;
-  };
-
   outputs = {
     self,
     nixpkgs,
     flake-utils,
   }:
     with flake-utils.lib;
-      eachSystem [system.x86_64-linux] (system: let
+      eachDefaultSystem (system: let
         pkgs = nixpkgs.legacyPackages.${system};
         hpkgs = pkgs.haskellPackages;
-        smtlib-backends = hpkgs.callCabal2nix "smtlib-backends" ./. {};
-        smtlib-backends-process = hpkgs.callCabal2nix "smtlib-backends-process" ./smtlib-backends-process {inherit smtlib-backends smtlib-backends-tests;};
-        smtlib-backends-tests = hpkgs.callCabal2nix "smtlib-backends-tests" ./smtlib-backends-tests {inherit smtlib-backends;};
-        smtlib-backends-z3 = hpkgs.callCabal2nix "smtlib-backends-z3" ./smtlib-backends-z3 {
+        regen = pkgs.writeShellApplication {
+          name = "regen";
+          ## coreutils for xargs
+          runtimeInputs = with pkgs; [cabal2nix findutils coreutils];
+          text = ''
+            set -o errexit
+            set -o pipefail
+            set -o xtrace
+            ## Nix automatically runs `shellcheck` on this script
+            # shellcheck disable=SC2016
+            find . -name '*.cabal' -print0 | \
+            xargs --null -I{} \
+            sh -c 'cabal2nix "$(dirname {})" > "$(basename --suffix='.cabal' {}).nix"'
```

Comment:
Why is it necessary to call `cabal2nix` inside `sh -c` here ?

Reply:
I don't think this explanation is correct, but I do think we are bound to go with `sh -c` if we use `xargs` here. `xargs` supports `{}` without `sh -c`. Try for instance `printf 'a\nb\nc' | xargs -I{} echo = {} =`. However, `xargs` is not a Shell and will only run a simple command. Since you want to use output redirection (to write into a `.nix` file with `>`), you need a Shell that will interpret it correctly.

I'll also add that using null separators in `find` and `xargs` is overkill and useless here because if file names contains spaces or newlines they will appear as several arguments to `dirname` and `basename` anyways. Might as well assume there are no spaces and get rid of `-print0` and `--null`.

A way to write this without `xargs` would be:
```sh
find . -name '*.cabal' | while read file; do
    cabal2nix "$(dirname "$file")" > "$(basename --suffix='.cabal' "$file")".nix
done
```
and if you want to use null-separators you can go for:
```sh
find . -name '*.cabal' -print0 | while IFS= read -d $'\0' file; do
```
and this one will not have the problem mentioned above (but I still think it's a bit overkill to try and protect ourselves from the case where someone creates a `.cabal` file whose name contains a newline).